### ISSUE TITLE
Stability 

### ISSUE THREAD

**Issue by jayp84847** [Luca96/dlib-minified-models#6]
Yet I  am working on project with dlib 68 face landmarks.dat file. But in many testing results seems very unstable. After some research I found dataset file. 

Here are some source where its very stable example. 

98 face landmarks: https://wywu.github.io/projects/LAB/WFLW.html

Very stable 68 landmarks: https://github.com/facebookresearch/supervision-by-registration

But can you guide me how can I make ".dat" file with them. Or can you make more stable ".dat" file and share with me?

**Comment by Luca96** [comment]
Hi, sorry for the delay.

You can download the official 68-point shape predictor from Dlib, which is more accurate and stable but larger and slower.

Or, you can create your own dat model. For the links you cited, if they provide images and landmark annotations, you have to just make an XML file and then train the model from that XML.

The XML file basically contains the path of each image in your dataset along which each single annotated point.

You can read this [article](https://medium.datadriveninvestor.com/training-alternative-dlib-shape-predictor-models-using-python-d1d8f8bd9f5c) for more information about that.

Hope it helps

**Comment by jayp84847** [comment]
Thank you for your very helpful reply. As you said I am already using official Dlib 68_face_landmarks.dat but I am also using face mesh in this project, so with this Dlib module face mesh seems vibrating. Like its good but not stable. I am very new at Dlib so I don't have much knowledge about it so can you share me any video about how to train dlib shape predictor.

**Comment by jayp84847** [comment]
Can we make ".dat" file like "68_face_landmarks.dat" using this dataset?

https://wywu.github.io/projects/LAB/WFLW.html

**Comment by Luca96** [comment]
Yes, there would be 98 landmarks instead of 68. 

The only difficulty is to parse the annotation file they provide. If I manage to find some time, I'll provide you with at least the xml file required by Dlib to train a shape predictor model.

**Comment by jayp84847** [comment]
Thank you so much for such a support.

**Comment by Luca96** [comment]
Good news!
I managed to train a model on WFLW. You can find it [here](https://github.com/Luca96/dlib-minified-models/tree/master/face_landmarks/wflw), along with training scripts in case you want to experiment further.

Let me know if there are still some problems, otherwise I will close the issue.

Cheers

**Comment by jayp84847** [comment]
Thank you for your such a great support. 

1> Did you trained with all 7000 images?
2> Its detecting good but not working with real time face tracking. 
3> Can I get your email I want share you some testing videos of my modules and want some help please.

**Comment by Luca96** [comment]
Thank you, so:
1. No, I trained only on the training set but you can easily edit the files I've provided and replicate the results on the whole dataset.
2. There are few things you can try: (a) use lightweight training options for the model at the cost of accuracy; you have to retrain the model, (b) use a faster face detector and/or downsample input images, (c) compile Dlib from source and enable latest CPU optimizations. You may want to use C++, instead of Python.
3. Well.. you can find my email on my GitHub profile, but I cannot guarantee you a response.

**Comment by jayp84847** [comment]
Thank you for your great response. One last think I want to share with you. Can this help to make nore stable and accurate face tracking module in terms of Dlib.

https://github.com/1adrianb/2D-and-3D-face-alignment

**Comment by jayp84847** [comment]
Hey luca,

I got some github where they describe how can we make 2d facial landmarks more stable and accurate. Can you please check it and let me know will it work with this dlib's 68_face_landmarks.dat. 

If yes will you help me please. 

Link : https://github.com/1adrianb/face-alignment

Training code and dataset link : https://github.com/1adrianb/2D-and-3D-face-alignment
